### DISCUSSION TITLE
I deleted red2.so and custom.css failed

### DISCUSSION THREAD

**Discussion by mstdnhomes** [mastodon/mastodon#18664] [Q&A] [answered]
🇶🇦 ❓ Hi. My site is https://mstdn.homes 

I run mastodon with docker image v3.5.3.

When I started to run a mastodon site, I was not good at Linux and accidentally expose my redis port to the external network. Then I was hacked. Later I closed the external port of redis. Recently I used clamav to scan and found this:
`/home/mastodon/mastodon/redis/red2.so: Unix.Trojan.Gafgyt-6981174-0 FOUND`

So I have to delete it. Now I just found that my custom.css didn't work. I also checked https://mstdn.homes/custom.css  
It shows as below even if I make the custom.css empty in the settings:

> We're sorry, but something went wrong on our end.

I don't know where to check. Could anyone help me please. Thank you.

For your information:

My server:
system: Oracle Linux 8 (Red Hat 8)
arch: aarch64
docker image: tootsuite/mastodon  v3.5.2  9248247a9a72  1.67GB

**Comment by nightpool** [comment]
I would recommend reinstalling Mastodon from scratch on a new server. Once you've been hacked, you have no idea what someone else could have possibly done to your machine. You should consider it fully compromised

**Reply by mstdnhomes** [reply]
Thanks. I run redis in docker and redis got hacked. I can reinstall the server with the database and previous media files. But I still can not use the **old redis data**.  SO is there any methods to rebuild the redis and fix the custom.css problem? Thanks

**Reply by mstdnhomes** [reply, marked as answer]
Hi. I reinstalled Mastodon from scratch on a new server ( with old **postgres14** and public **folder**). And the problem was solved! Thanks a lot
